File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/winforms/samples/printing/example5/vb/PrintForm.vb
'-----------------------------------------------------------------------
'  This file is part of the Microsoft .NET SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'-----------------------------------------------------------------------


Imports System
Imports System.ComponentModel
Imports System.Windows.Forms
Imports System.Drawing
Imports System.Drawing.Printing
Imports System.IO

Namespace Microsoft.Samples.WinForms.VB.PrintingExample5

    Public Class PrintForm
        Inherits System.Windows.Forms.Form

        Private storedPageSettings As PageSettings

        Public Sub New ()

            MyBase.New

            PrintForm = Me

            'This call is required by the Windows Forms Designer.
            InitializeComponent()

            'Wire up events for print buttons
            AddHandler printButton.Click, AddressOf printButton_Click
            AddHandler pageSetupButton.Click, AddressOf pageSetupButton_Click
            AddHandler printPreviewButton.Click, AddressOf printPreviewButton_Click

        End Sub

        'Event fired when the user presses the page setup button
        Private Sub pageSetupButton_Click(sender As object, e As System.EventArgs)

            Try
                Dim psDlg As New PageSetupDialog

                If (storedPageSettings Is Nothing) Then
                    storedPageSettings = new PageSettings()
                End If

                psDlg.PageSettings = storedPageSettings
                psDlg.ShowDialog
            Catch ex As Exception
                MessageBox.Show("An error occurred - " + ex.Message)
            End Try

        End Sub


        'Event fired when the user presses the print button
        Private Sub printButton_Click(sender As object, e As System.EventArgs)

            Try
                Dim streamToPrint As StreamReader = new StreamReader ("PrintMe.Txt")
                Try
                    'Assumes the default printer
                    Dim pd As TextFilePrintDocument = new TextFilePrintDocument(streamToPrint)

                    If Not (storedPageSettings Is Nothing) Then
                        pd.DefaultPageSettings = storedPageSettings
                    End If


                    Dim dlg As New PrintDialog()
                    dlg.Document = pd
                    Dim result As DialogResult = dlg.ShowDialog()

                    If (result = System.Windows.Forms.DialogResult.OK) Then
                        pd.Print()
                    End If

                Finally
                    streamToPrint.Close()
                End Try

            Catch ex As Exception
                MessageBox.Show("An error occurred printing the file - " + ex.Message)
            End Try

        End Sub


        'Event fired when the user presses the print preview button
        Private Sub printPreviewButton_Click(sender As object, e As System.EventArgs)

            Try
                Dim streamToPrint As StreamReader = new StreamReader ("PrintMe.Txt")
                Try
                    'Assumes the default printer
                    Dim pd As TextFilePrintDocument = new TextFilePrintDocument(streamToPrint)

                    If Not (storedPageSettings Is Nothing) Then
                        pd.DefaultPageSettings = storedPageSettings
                    End If

                    Dim dlg As New PrintPreviewDialog()
                    dlg.Document = pd
                    dlg.ShowDialog()

                Finally
                    streamToPrint.Close()
                End Try

            Catch ex As Exception
                MessageBox.Show("An error occurred - " + ex.Message)
            End Try

        End Sub


        'Form overrides dispose to clean up the component list.
        Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
            If disposing Then
                If Not (components Is Nothing) Then
                    components.Dispose()
                End If
            End If
            MyBase.Dispose(disposing)
        End Sub



#Region " Windows Form Designer generated code "

        'Required by the Windows Form Designer
        Private components As System.ComponentModel.Container
        Private WithEvents printButton As System.Windows.Forms.Button
        Private WithEvents pageSetupButton As System.Windows.Forms.Button
        Private WithEvents printPreviewButton As System.Windows.Forms.Button

        Private WithEvents PrintForm As System.Windows.Forms.Form

        'NOTE: The following procedure is required by the Windows Form Designer
        'It can be modified using the Windows Form Designer.
        'Do not modify it using the code editor.
        Private Sub InitializeComponent()
            Me.components = New System.ComponentModel.Container()
            Me.printPreviewButton = New System.Windows.Forms.Button()
            Me.pageSetupButton = New System.Windows.Forms.Button()
            Me.printButton = New System.Windows.Forms.Button()
            Me.Text = "Print Example 5"
            Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
            Me.ClientSize = New System.Drawing.Size(504, 381)

            printPreviewButton.ImageAlign = System.Drawing.ContentAlignment.MiddleLeft
            printPreviewButton.FlatStyle = System.Windows.Forms.FlatStyle.Flat
            printPreviewButton.Size = New System.Drawing.Size(136, 40)
            printPreviewButton.TabIndex = 2
            printPreviewButton.Location = New System.Drawing.Point(32, 210)
            printPreviewButton.Text = "Print Preview"

            pageSetupButton.ImageAlign = System.Drawing.ContentAlignment.MiddleLeft
            pageSetupButton.FlatStyle = System.Windows.Forms.FlatStyle.Flat
            pageSetupButton.Size = New System.Drawing.Size(136, 40)
            pageSetupButton.TabIndex = 1
            pageSetupButton.Location = New System.Drawing.Point(32, 160)
            pageSetupButton.Text = "Page Settings"

            printButton.ImageAlign = System.Drawing.ContentAlignment.MiddleLeft
            printButton.FlatStyle = System.Windows.Forms.FlatStyle.Flat
            printButton.Size = New System.Drawing.Size(136, 40)
            printButton.TabIndex = 0
            printButton.Location = New System.Drawing.Point(32, 112)
            printButton.Text = "Print the file"
            Me.Controls.Add(printButton)
            Me.Controls.Add(pageSetupButton)
            Me.Controls.Add(printPreviewButton)
        End Sub

#End Region

        'The main entry point for the application
        <STAThread()> Shared Sub Main()
            System.Windows.Forms.Application.Run(New PrintForm())
        End Sub

    End Class

End Namespace

File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/winforms/samples/printing/example5/cs/PrintForm.cs
//-----------------------------------------------------------------------
//  This file is part of the Microsoft .NET SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//-----------------------------------------------------------------------


namespace Microsoft.Samples.WinForms.Cs.PrintingExample5 {
    using System;
    using System.ComponentModel;
    using System.Windows.Forms;
    using System.Drawing;
    using System.Drawing.Printing;
    using System.IO;

    public class PrintForm : System.Windows.Forms.Form {
        /// <summary>
        ///    Required designer variable.
        /// </summary>
        private System.ComponentModel.Container components;
        protected internal System.Windows.Forms.Button printButton;
        protected internal System.Windows.Forms.Button pageSetupButton;
        protected internal System.Windows.Forms.Button printPreviewButton;

        private PageSettings storedPageSettings = null ;

        public PrintForm() {
            //
            // Required for Windows Form Designer support
            //
            InitializeComponent();

            // Wire up event handlers for print buttons
            printButton.Click += new System.EventHandler(printButton_Click);
            pageSetupButton.Click += new System.EventHandler(pageSetupButton_Click);
            printPreviewButton.Click += new System.EventHandler(printPreviewButton_Click);
        }

        //Event fired when the user presses the page setup button
        private void pageSetupButton_Click(object sender, EventArgs e) {

            try {
                PageSetupDialog psDlg = new PageSetupDialog() ;

                if (storedPageSettings == null) {
                    storedPageSettings =  new PageSettings();
                }

                psDlg.PageSettings = storedPageSettings ;
                psDlg.ShowDialog();

            } catch(Exception ex) {
                MessageBox.Show("An error occurred - " + ex.Message);
            }

        }


        //Event fired when the user presses the print button
        private void printButton_Click(object sender, EventArgs e) {
            try {

                StreamReader streamToPrint = new StreamReader ("PrintMe.Txt");
                try {
                    TextFilePrintDocument pd = new TextFilePrintDocument(streamToPrint); //Assumes the default printer

                    if (storedPageSettings != null) {
                        pd.DefaultPageSettings = storedPageSettings ;
                    }

                    PrintDialog dlg = new PrintDialog() ;
                    dlg.Document = pd;
                    DialogResult result = dlg.ShowDialog();

                    if (result == DialogResult.OK) {
                        pd.Print();
                    }

                } finally {
                    streamToPrint.Close() ;
                }

            } catch(Exception ex) {
                MessageBox.Show("An error occurred printing the file - " + ex.Message);
            }
        }


        //Event fired when the user presses the print preview button
        private void printPreviewButton_Click(object sender, EventArgs e) {
            try {

                StreamReader streamToPrint = new StreamReader ("PrintMe.Txt");
                try {
                    TextFilePrintDocument pd = new TextFilePrintDocument(streamToPrint); //Assumes the default printer

                    if (storedPageSettings != null) {
                        pd.DefaultPageSettings = storedPageSettings ;
                    }

                    PrintPreviewDialog dlg = new PrintPreviewDialog() ;
                    dlg.Document = pd;
                    dlg.ShowDialog();

                } finally {
                    streamToPrint.Close() ;
                }

            } catch(Exception ex) {
                MessageBox.Show("An error occurred attempting to preview the file to print - " + ex.Message);
            }

        }

        /// <summary>
        ///    Clean up any resources being used.
        /// </summary>
        protected override void Dispose(bool disposing)
        {
           if (disposing) {
                if (components != null) {
                    components.Dispose();
                }
           }
           base.Dispose(disposing);
        }

        /// <summary>
        ///    Required method for Designer support - do not modify
        ///    the contents of this method with the code editor.
        /// </summary>
        private void InitializeComponent()
        {
            this.components = new System.ComponentModel.Container ();
            this.printPreviewButton = new System.Windows.Forms.Button ();
            this.pageSetupButton = new System.Windows.Forms.Button ();
            this.printButton = new System.Windows.Forms.Button ();
            this.Text = "Print Example 5";
            this.AutoScaleBaseSize = new System.Drawing.Size (5, 13);
            this.ClientSize = new System.Drawing.Size (504, 381);
            printPreviewButton.ImageAlign = System.Drawing.ContentAlignment.MiddleLeft;
            printPreviewButton.FlatStyle = System.Windows.Forms.FlatStyle.Flat;
            printPreviewButton.Size = new System.Drawing.Size (136, 40);
            printPreviewButton.TabIndex = 0;
            printPreviewButton.Location = new System.Drawing.Point (32, 210);
            printPreviewButton.Text = "Print Preview";
            pageSetupButton.ImageAlign = System.Drawing.ContentAlignment.MiddleLeft;
            pageSetupButton.FlatStyle = System.Windows.Forms.FlatStyle.Flat;
            pageSetupButton.Size = new System.Drawing.Size (136, 40);
            pageSetupButton.TabIndex = 0;
            pageSetupButton.Location = new System.Drawing.Point (32, 160);
            pageSetupButton.Text = "Page Settings";
            printButton.ImageAlign = System.Drawing.ContentAlignment.MiddleLeft;
            printButton.FlatStyle = System.Windows.Forms.FlatStyle.Flat;
            printButton.Size = new System.Drawing.Size (136, 40);
            printButton.TabIndex = 0;
            printButton.Location = new System.Drawing.Point (32, 112);
            printButton.Text = "Print the file";
            this.Controls.Add (this.printButton);
            this.Controls.Add (this.pageSetupButton);
            this.Controls.Add (this.printPreviewButton);
        }

        /// <summary>
        /// The main entry point for the application.
        /// </summary>
        [STAThread]
        public static void Main(string[] args) {
            Application.Run(new PrintForm());
        }

    }
}






